krabs/krabs/ut.hpp

```diff
@@ -137,11 +147,10 @@ namespace krabs { namespace details {
             parameters.SourceId = provider.first;
```

Comment:
Very optional: you can use structured bindings to make all of the `provider.second` stuff below go away.
E.g.
```
	std::map<int, std::vector<int>> mapping;
	for (auto& [id, items] : mapping)
	{
		// id is the key
		// items is the value
	}
```
in this case `for (auto& [provider, settings] : provider_flags)` and then `provider.second` would be `settings` or whatever you name it.